lib/Exposed.ml

```diff
@@ -59,7 +59,8 @@ module Right = struct
     | {pcd_args= args; _} -> constructor_arguments args
 
   let type_declaration = function
-    | {ptype_attributes= _ :: _; _} -> false
+    | {ptype_attributes; _} when Ast.Ext_attrs.has_attrs ptype_attributes ->
```

Comment:
`Ext_attrs.has_attrs` is not right here as this checks whether an attribute will be present on the right of the item. We must only check `attrs_after`, otherwise the item will feel protected when it isn't.

For example, this generates invalid syntax:
```
[%foo
type[@foo] t = < .. >
]
```
(please add it to the testsuite)

Reply:
This does not surprise as I mostly rewrote this file in autopilote. Thanks for noticing this.